File: senior/MT1051/week12/HW34/HW34_109601003.py
# -*- coding: utf-8 -*-
"""
Date: 2023/12/11
HW: 34
Author: 林群賀
Student Number: 109601003
"""

import datetime

present_time = datetime.datetime.now()
print("Time:", present_time)

time_alert = datetime.datetime(2023, 12, 11, 9, 22, 00)
if (present_time > time_alert):
    print("Time is up!")

delta_time = datetime.timedelta(
    weeks=2, 
    days=3, hours=5, 
    minutes=15, 
    seconds=25
)
print("DeltaTime:", delta_time)

time_new = present_time + delta_time
print("Time after DelatTime is:", time_new)

File: other/MT1051/week12/HW34/HW34_109601003.go
/*
Created Date: 2023/12/11
HW: 34
Author: 林群賀
Student Number: 109601003
*/

package main

import (
	"fmt"
	"time"
)

func main() {
	presentTime := time.Now()
	fmt.Println("Time:", presentTime)

	timeAlert := time.Date(2023, 12, 11, 9, 22, 0, 0, time.UTC)
	if presentTime.After(timeAlert) {
		fmt.Println("Time is up!")
	}

	deltaTime := time.Date(0, 0, 0, 2*7+3, 5, 15, 25*1000000, time.UTC)
	fmt.Println("DeltaTime:", deltaTime)

	timeNew := presentTime.Add(deltaTime)
	fmt.Println("Time after DeltaTime is:", timeNew)
	fmt.Println("Time after DeltaTime is:", timeNew)
}
